source/agora/test/BanManager.d

```diff
@@ -43,13 +45,16 @@ unittest
     scope(failure) network.printLogs();
     network.waitForDiscovery();
 
-    // three nodes, two validators, and 1 non-validator
-    auto node_1 = network.nodes[0].client;
-    auto node_2 = network.nodes[1].client;
-    auto node_3 = network.nodes[2].client;  // non-validator
-    auto nodes = [node_1, node_2, node_3];
+    auto nodes = network.clients;
+    assert(nodes.length >= 3);
+    // The last node is non-validator, and the others are validators.
+    auto node_1 = nodes[0];     // validators
+    auto node_2 = nodes[1];     // validators
+    auto node_3 = nodes[$-1];   // non-validator
```

Comment:
```suggestion
    auto full_node = nodes[$-1];
```

That way it's explicit in the test, and also it doesn't matter which number it is anymore (node 3, 4, and so on).